### PR TITLE
Swift 4 Updates with Explicit Tuple Parameters

### PR DESCRIPTION
**Pull request by akwilliamson**
# What

- Updates compile error for non-implicit tuple parameter for an empty task with a success state.
- Resolves empty parameter deprecated warnings for `Task` execution class methods.

# Why

- To conform to the Swift 4 evolution of using explicit tuples described [here](https://github.com/apple/swift-evolution/blob/master/proposals/0029-remove-implicit-tuple-splat.md)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ branches:
     - master
 language: objective-c
 os: osx
-osx_image: xcode8
+osx_image: xcode9
```
Comment (akwilliamson) [inline .travis.yml:6]: Mistake on the commit message saying "Use Xcode 8" when it should have said "Use Xcode 9"

**Comment by z4r** [comment]
Any news on this? I’ve got the annoying issues on Code Coverage (now forbidden to release)

**Comment by akwilliamson** [comment]
@z4r Are you talking about test coverage on the PR? If that's the case, these changes shouldn't affect test coverage as it still sits above the 75% target.

**Comment by z4r** [comment]
Hi @akwilliamson,
no, I'm not talking about coverage of this module, but about this mail sent to me via App Store upload:

> Invalid Bundle - Disallowed LLVM instrumentation. Do not submit apps with LLVM profiling instrumentation or coverage collection enabled. Turn off LLVM profiling or code coverage, rebuild your app and resubmit the app.

It can be solved removing the **code coverage** check in the `test` tab

screen shot 2017-09-25 at 21 38 24

File: Sources/BoltsSwift/Task.swift [hunk 2]
```diff
@@ -93,7 +93,7 @@ public final class Task<TResult> {
     }
 
     class func emptyTask() -> Task<Void> {
-        return Task<Void>(state: .success())
+        return Task<Void>(state: .success(()))
```
Comment (rogerhu) [inline Sources/BoltsSwift/Task.swift:96]: +1 this fixed my issues too

**Comment by rogerhu** [comment]
Can we get this PR merged and deployed?  It is breaking Parse SDK on XCode 9 right now..

Thanks!

https://github.com/parse-community/Parse-SDK-iOS-OSX/issues/1193

**Pull request closed without merging**